### PR TITLE
feat(specs): add InvariantBaseTest for FeeAMM and StablecoinDEX

### PR DESCRIPTION
**Pull request by zerosnacks**
## Summary

Introduce shared test infrastructure for invariant testing of FeeAMM and StablecoinDEX.

## Changes

- **InvariantBaseTest.t.sol** (new): Shared base contract with:
  - Actor management (`_buildActors`, `_buildActorsWithApprovals`, `_selectActor`)
  - Token selection (`_selectToken`, `_selectBaseToken`, `_getTokenSymbol`)
  - Funding helpers (`_ensureFunds`, `_ensureFundsAll`)
  - Policy helpers (`_getPolicyId`, `_getPolicyAdmin`, `_isAuthorized`, `_setBlacklist`)
  - Logging utilities (`_log`, `_logContractBalances`, `_getActorIndex`)
  - Error handling (`_isKnownTIP20Error`, `_isKnownFeeAMMError`, `_isKnownDEXError`, etc.)

- **FeeAMM.t.sol**: Refactored to extend InvariantBaseTest, removing duplicated setup code

- **StablecoinDEX.t.sol**: Refactored to extend InvariantBaseTest, removing duplicated setup code

- **README.md**: Updated with FeeAMM and StablecoinDEX invariant documentation

## Context

This is the **first PR** in a series to split up the invariant test infrastructure. This PR keeps InvariantBaseTest minimal with only the helpers needed by FeeAMM and StablecoinDEX.

Subsequent PRs will add:
- AccountKeychain invariant tests
- Nonce invariant tests
- TIP20 invariant tests
- TIP20Factory invariant tests
- TIP403Registry invariant tests
- ValidatorConfig invariant tests

Each subsequent PR will extend InvariantBaseTest with additional helpers as needed.

## Testing

```bash
cd docs/specs && forge build --contracts test/invariants/
cd docs/specs && forge test --match-contract FeeAMMInvariantTest
cd docs/specs && forge test --match-contract StablecoinDEXInvariantTest
```

### CODE REVIEW

File: docs/specs/test/invariants/FeeAMM.t.sol [hunk 1]
```diff
@@ -630,19 +644,178 @@ contract FeeAMMInvariantTest is BaseTest {
         try amm.rebalanceSwap(userToken, validatorToken, amountOut, actor) returns (
             uint256 amountIn
         ) {
-            // TEMPO-AMM18: Small swaps should still pay >= theoretical rate
-            uint256 theoretical = (amountOut * N) / SCALE;
-            assertTrue(
-                amountIn >= theoretical, "TEMPO-AMM18: Small swap should pay >= theoretical rate"
+            // TEMPO-AMM10/18: Rebalance swap must follow exact formula: amountIn = floor(amountOut * N / SCALE) + 1
+            // This is the exact rounding-up formula that always favors the pool
+            uint256 expectedAmountIn = (amountOut * N) / SCALE + 1;
```
Comment (howydev) [inline docs/specs/test/invariants/FeeAMM.t.sol:649]: huh, this formula is not exactly right, since it adds 1 even in the case that `amountOut * N` is perfectly divisible by `SCALE`. In other places we do `(amountOut * N + scale - 1) / SCALE` 

Not that big of an issue since it doesn't cause exchange shortfall. I also suspect that the rust precompiles have the right implementation like above

File: docs/specs/test/invariants/StablecoinDEX.t.sol [hunk 2]
```diff
@@ -178,14 +140,69 @@ contract StablecoinDEXInvariantTest is BaseTest {
...
+    /// @notice TEMPO-DEX20: Test divisibility edge cases - when (base*price) % PRICE_SCALE == 0
+    function placeDivisibleBid(uint256 actorRnd, uint256 tickRnd, uint256 tokenRnd) external {
+        int16 tick = _ticks[tickRnd % _ticks.length];
+        address actor = _actors[actorRnd % _actors.length];
+        address token = address(_tokens[tokenRnd % _tokens.length]);
+        uint32 price = exchange.tickToPrice(tick);
+
+        // Calculate amount where (amount * price) % 100_000 == 0, above min order size
+        uint128 amount = uint128((100_000 / _gcd(uint256(price), 100_000)) * 10_000);
+        if (amount < 100_000_000) return;
```
Comment (howydev) [inline docs/specs/test/invariants/StablecoinDEX.t.sol:168]: would this trigger quite often? i think gcd(price, 100000) is always >=10 since ticks are every 10 units apart only
